tools/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/tools
 
-go 1.17
+go 1.18
```

Comment:
It looks like `honnef.co/go/tools` needs to be updated to v0.3.0 to support go1.18

Reply:
Thanks @tam7t  , i have updated few more libraries

Reply:
Fixed the static check on the test as well

Reply:
@tam7t i can rebase the commit if you need me to